pkg/autoscaler/autoscaler.go

```diff
@@ -352,14 +359,22 @@ func (a *Autoscaler) Scale(ctx context.Context, now time.Time) (int32, bool) {
 	return desiredPodCount, true
 }
 
-func (a *Autoscaler) rateLimited(desiredRate float64) float64 {
-	if desiredRate > a.Current().MaxScaleUpRate {
-		return a.Current().MaxScaleUpRate
+func (a *Autoscaler) podCountLimited(desiredPodCount, currentPodCount float64) float64 {
+	maxUpPodCount := a.Current().MaxScaleUpRate * currentPodCount
+	if desiredPodCount > maxUpPodCount {
```

Comment:
Usage of float64 has it's benefits :yum: 
```
return math.Min(dPC, mUPC)
```
and at that point you might just skip the `maxUppodCount` variable altogether and inline the computation.

Reply:
Done